src/test/crypto_tests.cpp

```diff
@@ -562,4 +564,198 @@ BOOST_AUTO_TEST_CASE(sha256d64)
     }
 }
 
+
+BOOST_AUTO_TEST_CASE(chacha20poly1305_openssh)
+{
+    struct chacha20_testvector {
+        uint8_t key[32];
+        uint8_t nonce[8];
+        uint8_t resulting_keystream[512];
+        int keystream_check_size;
+    };
+
+    struct poly1305_testvector {
+        uint8_t input[64];
+        int inputlen;
+        uint8_t key[64];
+        uint8_t resulting_tag[16];
+    };
+
+    /*
+       Testvectors have been taken from the draft RFC
+       https://tools.ietf.org/html/draft-agl-tls-chacha20poly1305-04#section-7
+    */
+
+    static const struct chacha20_testvector chacha20_testvectors[] = {
+        {{0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
```

Comment:
This would be much easier to verify against the RFC document if you parse a hex string.